docs/rancher/rancher-terraform.md

```diff
@@ -0,0 +1,188 @@
+    RANCHER_SERVER_URL="<RANCHER_SERVER_URL>" # Pure server URL like https://192.168.0.181:6443
+    RANCHER_ACCESS_KEY="<RANCHER_ACCESS_KEY>"
+    RANCHER_SECRET_KEY="<RANCHER_SECRET_KEY>"
+    HARVESTER_CLUSTER_ID="<HARVESTER_CLUSTER_ID>"
+    CLUSTER_NAME="rke2-demo"
+    curl -k -X POST ${RANCHER_SERVER_URL}/k8s/clusters/${HARVESTER_CLUSTER_ID}/v1/harvester/kubeconfig \
+    -H 'Content-Type: application/json' \
+    -u ${RANCHER_ACCESS_KEY}:${RANCHER_SECRET_KEY} \
+    -d '{"clusterRoleName": "harvesterhci.io:cloudprovider", "namespace": "default", "serviceAccountName": "'${CLUSTER_NAME}'"}' | xargs | sed 's/\\n/\n/g' > ${CLUSTER_NAME}-kubeconfig
+    ```
+
+1. Prepare a `provider.tf` file with the following content:
+
+    ```hcl
+    terraform {
+    required_providers {
+        rancher2 = {
+            source  = "rancher/rancher2"
+            version = "4.2.0"
+            }
+        }
+    }
+
+    # Configure the Rancher2 provider to admin
+    provider "rancher2" {
+        api_url    = "<api_url>" # API Endpoint on Account & API Keys page
+        access_key = "<access_key>"
+        secret_key = "<secret_key>"
+        insecure   = true # Set to true if the Rancher server uses a self-signed certificate
+    }
+    ```
+
+1. Prepare a `main.tf` file with the following content:
+
+    ```hcl
+    # Get imported harvester cluster info
+    data "rancher2_cluster_v2" "harv" {
+        name = "<harvester_cluster_name_in_rancher>"
+    }
+
+    # Create a new Cloud Credential for an imported Harvester cluster
+    resource "rancher2_cloud_credential" "harv-cred" {
+        name = "harv-cred"
+        harvester_credential_config {
+            cluster_id = data.rancher2_cluster_v2.harv.cluster_v1_id
+            cluster_type = "imported"
+            kubeconfig_content = data.rancher2_cluster_v2.harv.kube_config
+        }
+    }
+
+    # Create a new rancher2 machine config v2 using harvester node_driver
+    resource "rancher2_machine_config_v2" "rke2-machine" {
+        generate_name = "rke2-machine"
+        harvester_config {
+            vm_namespace = "default"
+            cpu_count = "2"
+            memory_size = "4"
+            disk_info = <<EOF
+            {
+                "disks": [{
+                    "imageName": "default/<vmimage-name>",
+                    "size": 15,
+                    "bootOrder": 1
+                }]
+            }
+            EOF
+            network_info = <<EOF
+            {
+                "interfaces": [{
+                    "networkName": "default/<network-name>"
+                }]
+            }
+            EOF
+            ssh_user = "<ssh_user>"
+            user_data = <<EOF
+            package_update: true
+            packages:
+            - qemu-guest-agent
+            - iptables
+            runcmd:
+            - - systemctl
+                - enable
+                - '--now'
+                - qemu-guest-agent.service
+            password: test
```

Comment:
Let's remove this from the example, it's good for debugging purposes but not for production.

Reply:
Yes, removed it. Thanks.